File: backend/app/routes/countries.py
from fastapi import APIRouter

router = APIRouter(tags=["countries"])


@router.get("/countries/{iso3}/profile")
async def get_country_profile(iso3: str):
    return {}


@router.get("/countries/{iso3}/partners")
async def get_country_partners(iso3: str):
    return []

File: frontend/src/api/countries.js
import api from "./client";

export async function fetchCountryProfile(iso3, year, commodity, signal) {
  const { data } = await api.get(`/countries/${iso3}/profile`, {
    params: { year, commodity },
    signal,
  });
  return data;
}

export async function fetchCountryPartners(iso3, year, commodity, flowType, signal) {
  const { data } = await api.get(`/countries/${iso3}/partners`, {
    params: { year, commodity, flow_type: flowType },
    signal,
  });
  return data;
}
